packages/react-select/src/components/Select/useSelectStyles.ts

```diff
@@ -12,36 +13,211 @@ export const selectClassNames: SlotClassNames<SelectSlots> = {
+    },
+  },
+});
+
+const useSelectStyles = makeStyles({
+  base: {
+    appearance: 'none',
+    ...shorthands.border('1px', 'solid', 'transparent'),
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+    boxShadow: 'none',
+    boxSizing: 'border-box',
+    color: tokens.colorNeutralForeground1,
+    flexGrow: 1,
+    fontFamily: tokens.fontFamilyBase,
+
+    ':focus-visible': {
+      outlineWidth: '2px',
+      outlineStyle: 'solid',
+      outlineColor: 'transparent',
+    },
+  },
+  disabled: {
+    backgroundColor: tokens.colorTransparentBackground,
+    color: tokens.colorNeutralForegroundDisabled,
+    cursor: 'not-allowed',
+  },
+  small: {
+    height: fieldHeights.small,
+    ...shorthands.padding('0', horizontalSpacing.sNudge),
+    ...contentSizes.caption1,
+  },
+  medium: {
+    height: fieldHeights.medium,
+    ...shorthands.padding('0', horizontalSpacing.mNudge),
+    ...contentSizes.body1,
+  },
+  large: {
+    height: fieldHeights.large,
+    ...shorthands.padding('0', horizontalSpacing.m),
+    ...contentSizes[400],
+  },
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border('1px', 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+  },
+  underline: {
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.borderBottom('1px', 'solid', tokens.colorNeutralStrokeAccessible),
+    ...shorthands.borderRadius(0),
   },
-  select: {
-    // TODO: add styles
+  filledLighter: {
+    backgroundColor: tokens.colorNeutralBackground1,
   },
-  selectDisabled: {
-    // TODO: add styles
+  filledDarker: {
+    backgroundColor: tokens.colorNeutralBackground3,
+  },
+});
+
+const useIconStyles = makeStyles({
+  icon: {
+    boxSizing: 'border-box',
+    color: tokens.colorNeutralStrokeAccessible,
+    display: 'block',
+    position: 'absolute',
+    right: '0',
+    pointerEvents: 'none',
+
+    // the SVG must have display: block for accurate positioning
+    // otherwise an extra inline space is inserted after the svg element
+    '& svg': {
+      display: 'block',
+    },
+  },
+  small: {
+    fontSize: iconSizes.small,
+    height: iconSizes.small,
+    paddingRight: horizontalSpacing.sNudge,
+    paddingLeft: horizontalSpacing.xxs,
+    width: iconSizes.small,
+  },
+  medium: {
+    fontSize: iconSizes.medium,
+    height: iconSizes.medium,
+    paddingRight: horizontalSpacing.mNudge,
+    paddingLeft: horizontalSpacing.xxs,
+    width: iconSizes.medium,
+  },
+  large: {
+    fontSize: iconSizes.large,
+    height: iconSizes.large,
+    paddingRight: horizontalSpacing.m,
+    paddingLeft: horizontalSpacing.sNudge,
+    width: iconSizes.large,
   },
 });
 
 /**
  * Apply styling to the Select slots based on the state
  */
 export const useSelectStyles_unstable = (state: SelectState): SelectState => {
+  const { size = 'medium', appearance = 'outline' } = state;
```

Comment:
might make more sense to assign these in the state hook ?

Reply:
looks like they already are, removing the defaults here :)